From 25b3524ec1fd000bab823d4e67cb25179f71aedd Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 21 Feb 2022 10:05:39 +1100
Subject: [PATCH 4/8] ACTUALLY use external absl

make include order consistent:
- own header
- std facilities
- external code
- internal code
---
 src/CMakeLists.txt                    |  6 ------
 src/bpe_model.cc                      |  3 ++-
 src/bpe_model_trainer.cc              |  7 ++++---
 src/bpe_model_trainer.h               |  3 ++-
 src/bpe_model_trainer_test.cc         |  5 +++--
 src/builder.cc                        |  9 +++++----
 src/builder.h                         |  3 ++-
 src/builder_test.cc                   |  3 ++-
 src/char_model_trainer_test.cc        |  5 +++--
 src/common.h                          |  3 ++-
 src/compile_charsmap_main.cc          |  5 +++--
 src/error.cc                          |  4 ++--
 src/filesystem.cc                     |  3 ++-
 src/filesystem.h                      |  3 ++-
 src/filesystem_test.cc                |  3 ++-
 src/init.h                            |  5 +++--
 src/model_factory.cc                  |  3 ++-
 src/model_interface.cc                |  5 +++--
 src/model_interface.h                 |  5 +++--
 src/model_interface_test.cc           |  3 ++-
 src/normalizer.cc                     |  9 +++++----
 src/normalizer.h                      |  3 ++-
 src/pretokenizer_for_training.cc      |  3 ++-
 src/pretokenizer_for_training.h       |  3 ++-
 src/pretokenizer_for_training_test.cc |  8 +++++---
 src/sentencepiece_processor.cc        | 17 +++++++++--------
 src/sentencepiece_processor.h         |  9 ++-------
 src/sentencepiece_processor_test.cc   |  9 +++++----
 src/sentencepiece_trainer.cc          | 13 +++++++------
 src/sentencepiece_trainer_test.cc     |  3 ++-
 src/spec_parser.h                     |  5 +++--
 src/spm_decode_main.cc                |  5 +++--
 src/spm_encode_main.cc                |  9 +++++----
 src/spm_export_vocab_main.cc          |  3 ++-
 src/spm_normalize_main.cc             |  3 ++-
 src/spm_train_main.cc                 |  9 +++++----
 src/testharness.cc                    |  3 ++-
 src/testharness.h                     |  7 ++++---
 src/trainer_factory.cc                |  3 ++-
 src/trainer_interface.cc              | 19 ++++++++++---------
 src/trainer_interface.h               |  3 ++-
 src/trainer_interface_test.cc         |  5 +++--
 src/unicode_script.cc                 |  3 ++-
 src/unicode_script_map.h              |  4 +++-
 src/unicode_script_test.cc            |  3 ++-
 src/unigram_model.cc                  |  9 +++++----
 src/unigram_model_test.cc             |  5 +++--
 src/unigram_model_trainer.cc          |  9 +++++----
 src/unigram_model_trainer.h           |  3 ++-
 src/unigram_model_trainer_test.cc     |  5 +++--
 src/util.h                            |  3 ++-
 src/util_test.cc                      |  3 ++-
 src/word_model_trainer.cc             |  5 +++--
 src/word_model_trainer_test.cc        |  5 +++--
 54 files changed, 168 insertions(+), 126 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6dfd76d..d30a4c2 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -89,12 +89,6 @@ endif()
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../third_party)
 
-if (MSVC)
-  add_definitions("/D_USE_INTERNAL_STRING_VIEW")
-else()
-  add_definitions("-D_USE_INTERNAL_STRING_VIEW")
-endif()
-
 set(SPM_SRCS
   ${PROTOBUF_LITE_SRCS}
   ${SPM_PROTO_HDRS}
diff --git a/src/bpe_model.cc b/src/bpe_model.cc
index bc7ada1..6cb5ab4 100644
--- a/src/bpe_model.cc
+++ b/src/bpe_model.cc
@@ -21,8 +21,9 @@
 #include <utility>
 #include <vector>
 
+#include "absl/container/flat_hash_map.h"
+
 #include "freelist.h"
-#include "third_party/absl/container/flat_hash_map.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/bpe_model_trainer.cc b/src/bpe_model_trainer.cc
index de86f14..42d737a 100644
--- a/src/bpe_model_trainer.cc
+++ b/src/bpe_model_trainer.cc
@@ -19,10 +19,11 @@
 #include <unordered_set>
 #include <vector>
 
+#include "absl/container/flat_hash_set.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_replace.h"
+
 #include "pretokenizer_for_training.h"
-#include "third_party/absl/container/flat_hash_set.h"
-#include "third_party/absl/strings/str_join.h"
-#include "third_party/absl/strings/str_replace.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/bpe_model_trainer.h b/src/bpe_model_trainer.h
index 2fdfb9c..980a42c 100644
--- a/src/bpe_model_trainer.h
+++ b/src/bpe_model_trainer.h
@@ -21,8 +21,9 @@
 #include <string>
 #include <vector>
 
+#include "absl/container/flat_hash_map.h"
+
 #include "sentencepiece_model.pb.h"
-#include "third_party/absl/container/flat_hash_map.h"
 #include "trainer_interface.h"
 
 namespace sentencepiece {
diff --git a/src/bpe_model_trainer_test.cc b/src/bpe_model_trainer_test.cc
index 173eb9c..f977fd0 100644
--- a/src/bpe_model_trainer_test.cc
+++ b/src/bpe_model_trainer_test.cc
@@ -15,13 +15,14 @@
 #include <string>
 #include <vector>
 
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+
 #include "bpe_model_trainer.h"
 #include "filesystem.h"
 #include "sentencepiece_processor.h"
 #include "sentencepiece_trainer.h"
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_join.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/builder.cc b/src/builder.cc
index 822f6fc..f804e2f 100644
--- a/src/builder.cc
+++ b/src/builder.cc
@@ -18,11 +18,12 @@
 #include <functional>
 #include <utility>
 
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_replace.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/strip.h"
+
 #include "filesystem.h"
-#include "third_party/absl/strings/str_join.h"
-#include "third_party/absl/strings/str_replace.h"
-#include "third_party/absl/strings/str_split.h"
-#include "third_party/absl/strings/strip.h"
 
 #ifdef ENABLE_NFKC_COMPILE
 #include <unicode/errorcode.h>
diff --git a/src/builder.h b/src/builder.h
index 094da72..8cd4d96 100644
--- a/src/builder.h
+++ b/src/builder.h
@@ -19,10 +19,11 @@
 #include <string>
 #include <vector>
 
+#include "absl/strings/string_view.h"
+
 #include "common.h"
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_processor.h"
-#include "third_party/absl/strings/string_view.h"
 
 namespace sentencepiece {
 namespace normalizer {
diff --git a/src/builder_test.cc b/src/builder_test.cc
index 4acb7b3..f586fba 100644
--- a/src/builder_test.cc
+++ b/src/builder_test.cc
@@ -12,13 +12,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.!
 
+#include "absl/strings/str_cat.h"
+
 #include "builder.h"
 #include "common.h"
 #include "filesystem.h"
 #include "normalizer.h"
 #include "sentencepiece_trainer.h"
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/char_model_trainer_test.cc b/src/char_model_trainer_test.cc
index 8c2e4b7..b49cea1 100644
--- a/src/char_model_trainer_test.cc
+++ b/src/char_model_trainer_test.cc
@@ -15,12 +15,13 @@
 #include <string>
 #include <vector>
 
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+
 #include "char_model_trainer.h"
 #include "filesystem.h"
 #include "sentencepiece_processor.h"
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_join.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/common.h b/src/common.h
index ef5546d..5245004 100644
--- a/src/common.h
+++ b/src/common.h
@@ -25,8 +25,9 @@
 #include <utility>
 #include <vector>
 
+#include "absl/strings/string_view.h"
+
 #include "config.h"
-#include "third_party/absl/strings/string_view.h"
 
 #if defined(_WIN32) && !defined(__CYGWIN__)
 #define OS_WIN
diff --git a/src/compile_charsmap_main.cc b/src/compile_charsmap_main.cc
index da15328..23e5ef8 100644
--- a/src/compile_charsmap_main.cc
+++ b/src/compile_charsmap_main.cc
@@ -18,12 +18,13 @@
 #include <sstream>
 #include <string>
 
+#include "absl/flags/flag.h"
+#include "absl/strings/string_view.h"
+
 #include "builder.h"
 #include "filesystem.h"
 #include "init.h"
 #include "sentencepiece_processor.h"
-#include "third_party/absl/flags/flag.h"
-#include "third_party/absl/strings/string_view.h"
 
 using sentencepiece::normalizer::Builder;
 
diff --git a/src/error.cc b/src/error.cc
index d3792dc..19ef6f3 100644
--- a/src/error.cc
+++ b/src/error.cc
@@ -21,8 +21,8 @@
 #ifdef _USE_EXTERNAL_ABSL
 // Naive workaround to define minloglevel on external absl package.
 // We want to define them in other cc file.
-#include "third_party/absl/flags/flag.h"
-#include "third_party/absl/flags/parse.h"
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
 ABSL_FLAG(int32, minloglevel, 0,
           "Messages logged at a lower level than this don't actually.");
 #endif
diff --git a/src/filesystem.cc b/src/filesystem.cc
index 833c8f7..baf8ad9 100644
--- a/src/filesystem.cc
+++ b/src/filesystem.cc
@@ -14,8 +14,9 @@
 
 #include <iostream>
 
+#include "absl/memory/memory.h"
+
 #include "filesystem.h"
-#include "third_party/absl/memory/memory.h"
 #include "util.h"
 
 #if defined(OS_WIN) && defined(UNICODE) && defined(_UNICODE)
diff --git a/src/filesystem.h b/src/filesystem.h
index e572b4b..50e9fa8 100644
--- a/src/filesystem.h
+++ b/src/filesystem.h
@@ -21,9 +21,10 @@
 #include <memory>
 #include <string>
 
+#include "absl/strings/string_view.h"
+
 #include "common.h"
 #include "sentencepiece_processor.h"
-#include "third_party/absl/strings/string_view.h"
 
 namespace sentencepiece {
 namespace filesystem {
diff --git a/src/filesystem_test.cc b/src/filesystem_test.cc
index 790e756..3c9fbdc 100644
--- a/src/filesystem_test.cc
+++ b/src/filesystem_test.cc
@@ -12,9 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.!
 
+#include "absl/strings/str_cat.h"
+
 #include "filesystem.h"
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/init.h b/src/init.h
index 6ae047e..1f4c292 100644
--- a/src/init.h
+++ b/src/init.h
@@ -15,9 +15,10 @@
 #ifndef INIT_H_
 #define INIT_H_
 
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+
 #include "common.h"
-#include "third_party/absl/flags/flag.h"
-#include "third_party/absl/flags/parse.h"
 
 #ifdef _USE_EXTERNAL_PROTOBUF
 #include "google/protobuf/message_lite.h"
diff --git a/src/model_factory.cc b/src/model_factory.cc
index be99501..a7ccf0f 100644
--- a/src/model_factory.cc
+++ b/src/model_factory.cc
@@ -12,10 +12,11 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.!
 
+#include "absl/memory/memory.h"
+
 #include "bpe_model.h"
 #include "char_model.h"
 #include "model_factory.h"
-#include "third_party/absl/memory/memory.h"
 #include "unigram_model.h"
 #include "word_model.h"
 
diff --git a/src/model_interface.cc b/src/model_interface.cc
index c49be1e..3ab6a35 100644
--- a/src/model_interface.cc
+++ b/src/model_interface.cc
@@ -14,10 +14,11 @@
 
 #include <algorithm>
 
+#include "absl/memory/memory.h"
+#include "absl/strings/str_format.h"
+
 #include "model_interface.h"
 #include "sentencepiece_model.pb.h"
-#include "third_party/absl/memory/memory.h"
-#include "third_party/absl/strings/str_format.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/model_interface.h b/src/model_interface.h
index 06e9243..b22b671 100644
--- a/src/model_interface.h
+++ b/src/model_interface.h
@@ -21,12 +21,13 @@
 #include <utility>
 #include <vector>
 
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/string_view.h"
+
 #include "common.h"
 #include "normalizer.h"
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_processor.h"
-#include "third_party/absl/container/flat_hash_map.h"
-#include "third_party/absl/strings/string_view.h"
 #include "third_party/darts_clone/darts.h"
 #include "util.h"
 
diff --git a/src/model_interface_test.cc b/src/model_interface_test.cc
index 09e41d3..865e427 100644
--- a/src/model_interface_test.cc
+++ b/src/model_interface_test.cc
@@ -14,9 +14,10 @@
 
 #include "model_interface.h"
 
+#include "absl/container/flat_hash_map.h"
+
 #include "model_factory.h"
 #include "testharness.h"
-#include "third_party/absl/container/flat_hash_map.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/normalizer.cc b/src/normalizer.cc
index 2ab8084..e0af25b 100644
--- a/src/normalizer.cc
+++ b/src/normalizer.cc
@@ -17,11 +17,12 @@
 #include <utility>
 #include <vector>
 
+#include "absl/memory/memory.h"
+#include "absl/strings/match.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+
 #include "common.h"
-#include "third_party/absl/memory/memory.h"
-#include "third_party/absl/strings/match.h"
-#include "third_party/absl/strings/string_view.h"
-#include "third_party/absl/strings/strip.h"
 #include "third_party/darts_clone/darts.h"
 #include "util.h"
 
diff --git a/src/normalizer.h b/src/normalizer.h
index c79813c..50b5108 100644
--- a/src/normalizer.h
+++ b/src/normalizer.h
@@ -21,10 +21,11 @@
 #include <utility>
 #include <vector>
 
+#include "absl/strings/string_view.h"
+
 #include "common.h"
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_processor.h"
-#include "third_party/absl/strings/string_view.h"
 #include "third_party/darts_clone/darts.h"
 
 namespace sentencepiece {
diff --git a/src/pretokenizer_for_training.cc b/src/pretokenizer_for_training.cc
index d4f492c..03df377 100644
--- a/src/pretokenizer_for_training.cc
+++ b/src/pretokenizer_for_training.cc
@@ -11,11 +11,12 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.!
+
 #include "pretokenizer_for_training.h"
 
 #include <string>
 
-#include "third_party/absl/strings/str_replace.h"
+#include "absl/strings/str_replace.h"
 
 namespace sentencepiece {
 namespace pretokenizer {
diff --git a/src/pretokenizer_for_training.h b/src/pretokenizer_for_training.h
index fa54f95..1ec5dfd 100644
--- a/src/pretokenizer_for_training.h
+++ b/src/pretokenizer_for_training.h
@@ -18,10 +18,11 @@
 #include <memory>
 #include <string>
 
+#include "absl/strings/string_view.h"
+
 #include "common.h"
 #include "sentencepiece.pb.h"
 #include "sentencepiece_processor.h"
-#include "third_party/absl/strings/string_view.h"
 
 namespace sentencepiece {
 namespace pretokenizer {
diff --git a/src/pretokenizer_for_training_test.cc b/src/pretokenizer_for_training_test.cc
index 99db0c5..a968b57 100644
--- a/src/pretokenizer_for_training_test.cc
+++ b/src/pretokenizer_for_training_test.cc
@@ -11,12 +11,14 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.!
+
 #include "pretokenizer_for_training.h"
 
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_split.h"
+
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_join.h"
-#include "third_party/absl/strings/str_split.h"
 #include "trainer_interface.h"
 
 namespace sentencepiece {
diff --git a/src/sentencepiece_processor.cc b/src/sentencepiece_processor.cc
index f0df2f6..5d80844 100644
--- a/src/sentencepiece_processor.cc
+++ b/src/sentencepiece_processor.cc
@@ -18,20 +18,21 @@
 #include <set>
 #include <utility>
 
+#include "absl/memory/memory.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_replace.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+
 #include "common.h"
 #include "filesystem.h"
 #include "model_factory.h"
 #include "model_interface.h"
 #include "normalizer.h"
 #include "sentencepiece.pb.h"
-#include "third_party/absl/memory/memory.h"
-#include "third_party/absl/strings/numbers.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_join.h"
-#include "third_party/absl/strings/str_replace.h"
-#include "third_party/absl/strings/str_split.h"
-#include "third_party/absl/strings/string_view.h"
-#include "third_party/absl/strings/strip.h"
 #include "unigram_model.h"
 #include "util.h"
 
diff --git a/src/sentencepiece_processor.h b/src/sentencepiece_processor.h
index 14b1e8c..b81cff5 100644
--- a/src/sentencepiece_processor.h
+++ b/src/sentencepiece_processor.h
@@ -15,19 +15,14 @@
 #ifndef SENTENCEPIECE_PROCESSOR_H_
 #define SENTENCEPIECE_PROCESSOR_H_
 
+#include "absl/strings/string_view.h"
+
 #include <cstring>
 #include <memory>
 #include <string>
-#include <string_view>
 #include <utility>
 #include <vector>
 
-#ifndef SWIG
-namespace absl {
-using std::string_view;
-}  // namespace absl
-#endif  // SWIG
-
 namespace sentencepiece {
 namespace util {
 
diff --git a/src/sentencepiece_processor_test.cc b/src/sentencepiece_processor_test.cc
index f05dc5d..4077c65 100644
--- a/src/sentencepiece_processor_test.cc
+++ b/src/sentencepiece_processor_test.cc
@@ -16,6 +16,11 @@
 
 #include <utility>
 
+#include "absl/container/flat_hash_map.h"
+#include "absl/memory/memory.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
+
 #include "builder.h"
 #include "filesystem.h"
 #include "model_interface.h"
@@ -24,10 +29,6 @@
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_trainer.h"
 #include "testharness.h"
-#include "third_party/absl/container/flat_hash_map.h"
-#include "third_party/absl/memory/memory.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/string_view.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/sentencepiece_trainer.cc b/src/sentencepiece_trainer.cc
index b9fe64f..ae8a7af 100644
--- a/src/sentencepiece_trainer.cc
+++ b/src/sentencepiece_trainer.cc
@@ -15,6 +15,13 @@
 #include <string>
 #include <vector>
 
+#include "absl/flags/flag.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+
 #include "builder.h"
 #include "common.h"
 #include "normalizer.h"
@@ -22,12 +29,6 @@
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_trainer.h"
 #include "spec_parser.h"
-#include "third_party/absl/flags/flag.h"
-#include "third_party/absl/strings/numbers.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_split.h"
-#include "third_party/absl/strings/string_view.h"
-#include "third_party/absl/strings/strip.h"
 #include "trainer_factory.h"
 #include "util.h"
 
diff --git a/src/sentencepiece_trainer_test.cc b/src/sentencepiece_trainer_test.cc
index e44e66b..0bb5aab 100644
--- a/src/sentencepiece_trainer_test.cc
+++ b/src/sentencepiece_trainer_test.cc
@@ -12,11 +12,12 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.!
 
+#include "absl/strings/str_cat.h"
+
 #include "filesystem.h"
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_trainer.h"
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/spec_parser.h b/src/spec_parser.h
index c5f0582..3871621 100644
--- a/src/spec_parser.h
+++ b/src/spec_parser.h
@@ -18,9 +18,10 @@
 #include <string>
 #include <vector>
 
+#include "absl/strings/ascii.h"
+#include "absl/strings/str_split.h"
+
 #include "sentencepiece_processor.h"
-#include "third_party/absl/strings/ascii.h"
-#include "third_party/absl/strings/str_split.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/spm_decode_main.cc b/src/spm_decode_main.cc
index bc49bd3..32fbb76 100644
--- a/src/spm_decode_main.cc
+++ b/src/spm_decode_main.cc
@@ -16,13 +16,14 @@
 #include <string>
 #include <vector>
 
+#include "absl/flags/flag.h"
+#include "absl/strings/str_split.h"
+
 #include "common.h"
 #include "filesystem.h"
 #include "init.h"
 #include "sentencepiece.pb.h"
 #include "sentencepiece_processor.h"
-#include "third_party/absl/flags/flag.h"
-#include "third_party/absl/strings/str_split.h"
 #include "util.h"
 
 ABSL_FLAG(std::string, model, "", "model file name");
diff --git a/src/spm_encode_main.cc b/src/spm_encode_main.cc
index 2fbb850..02def40 100644
--- a/src/spm_encode_main.cc
+++ b/src/spm_encode_main.cc
@@ -16,15 +16,16 @@
 #include <string>
 #include <vector>
 
+#include "absl/container/flat_hash_map.h"
+#include "absl/flags/flag.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+
 #include "common.h"
 #include "filesystem.h"
 #include "init.h"
 #include "sentencepiece.pb.h"
 #include "sentencepiece_processor.h"
-#include "third_party/absl/container/flat_hash_map.h"
-#include "third_party/absl/flags/flag.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_join.h"
 #include "trainer_interface.h"
 
 ABSL_FLAG(std::string, model, "", "model file name");
diff --git a/src/spm_export_vocab_main.cc b/src/spm_export_vocab_main.cc
index e5b97df..d0aea7d 100644
--- a/src/spm_export_vocab_main.cc
+++ b/src/spm_export_vocab_main.cc
@@ -14,12 +14,13 @@
 
 #include <sstream>
 
+#include "absl/flags/flag.h"
+
 #include "common.h"
 #include "filesystem.h"
 #include "init.h"
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_processor.h"
-#include "third_party/absl/flags/flag.h"
 
 ABSL_FLAG(std::string, output, "", "Output filename");
 ABSL_FLAG(std::string, model, "", "input model file name");
diff --git a/src/spm_normalize_main.cc b/src/spm_normalize_main.cc
index 39f3ef9..7352f63 100644
--- a/src/spm_normalize_main.cc
+++ b/src/spm_normalize_main.cc
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.!
 
+#include "absl/flags/flag.h"
+
 #include "builder.h"
 #include "common.h"
 #include "filesystem.h"
@@ -21,7 +23,6 @@
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_processor.h"
 #include "sentencepiece_trainer.h"
-#include "third_party/absl/flags/flag.h"
 
 ABSL_FLAG(std::string, model, "", "Model file name");
 ABSL_FLAG(bool, use_internal_normalization, false,
diff --git a/src/spm_train_main.cc b/src/spm_train_main.cc
index 34369cd..08c519f 100644
--- a/src/spm_train_main.cc
+++ b/src/spm_train_main.cc
@@ -14,14 +14,15 @@
 
 #include <map>
 
+#include "absl/flags/flag.h"
+#include "absl/strings/ascii.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_split.h"
+
 #include "filesystem.h"
 #include "init.h"
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_trainer.h"
-#include "third_party/absl/flags/flag.h"
-#include "third_party/absl/strings/ascii.h"
-#include "third_party/absl/strings/str_join.h"
-#include "third_party/absl/strings/str_split.h"
 #include "util.h"
 
 using sentencepiece::NormalizerSpec;
diff --git a/src/testharness.cc b/src/testharness.cc
index f6b1efe..6769cd8 100644
--- a/src/testharness.cc
+++ b/src/testharness.cc
@@ -25,8 +25,9 @@
 #include <string>
 #include <vector>
 
+#include "absl/strings/str_cat.h"
+
 #include "common.h"
-#include "third_party/absl/strings/str_cat.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/testharness.h b/src/testharness.h
index 9879b06..8de9c5c 100644
--- a/src/testharness.h
+++ b/src/testharness.h
@@ -20,10 +20,11 @@
 #include <sstream>
 #include <string>
 
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "absl/strings/string_view.h"
+
 #include "common.h"
-#include "third_party/absl/flags/flag.h"
-#include "third_party/absl/flags/parse.h"
-#include "third_party/absl/strings/string_view.h"
 
 ABSL_DECLARE_FLAG(std::string, test_tmpdir);
 ABSL_DECLARE_FLAG(std::string, test_srcdir);
diff --git a/src/trainer_factory.cc b/src/trainer_factory.cc
index d1d2541..34abe3f 100644
--- a/src/trainer_factory.cc
+++ b/src/trainer_factory.cc
@@ -12,9 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.!
 
+#include "absl/memory/memory.h"
+
 #include "bpe_model_trainer.h"
 #include "char_model_trainer.h"
-#include "third_party/absl/memory/memory.h"
 #include "trainer_factory.h"
 #include "unigram_model_trainer.h"
 #include "word_model_trainer.h"
diff --git a/src/trainer_interface.cc b/src/trainer_interface.cc
index 968f7b9..27cec4f 100644
--- a/src/trainer_interface.cc
+++ b/src/trainer_interface.cc
@@ -22,21 +22,22 @@
 #include <utility>
 #include <vector>
 
+#include "absl/container/flat_hash_map.h"
+#include "absl/memory/memory.h"
+#include "absl/random/distributions.h"
+#include "absl/random/random.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_split.h"
+
 #include "filesystem.h"
 #include "model_factory.h"
 #include "model_interface.h"
 #include "normalizer.h"
 #include "sentencepiece_processor.h"
 #include "sentencepiece_trainer.h"
-#include "third_party/absl/container/flat_hash_map.h"
-#include "third_party/absl/memory/memory.h"
-#include "third_party/absl/random/distributions.h"
-#include "third_party/absl/random/random.h"
-#include "third_party/absl/strings/numbers.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_format.h"
-#include "third_party/absl/strings/str_join.h"
-#include "third_party/absl/strings/str_split.h"
 #include "unicode_script.h"
 #include "util.h"
 
diff --git a/src/trainer_interface.h b/src/trainer_interface.h
index 8d625a9..58425c5 100644
--- a/src/trainer_interface.h
+++ b/src/trainer_interface.h
@@ -22,12 +22,13 @@
 #include <utility>
 #include <vector>
 
+#include "absl/container/flat_hash_map.h"
+
 #include "common.h"
 #include "filesystem.h"
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_processor.h"
 #include "sentencepiece_trainer.h"
-#include "third_party/absl/container/flat_hash_map.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/trainer_interface_test.cc b/src/trainer_interface_test.cc
index feb970f..75e9f54 100644
--- a/src/trainer_interface_test.cc
+++ b/src/trainer_interface_test.cc
@@ -16,10 +16,11 @@
 
 #include <utility>
 
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_format.h"
+
 #include "filesystem.h"
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_format.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/unicode_script.cc b/src/unicode_script.cc
index 583dc30..5790566 100644
--- a/src/unicode_script.cc
+++ b/src/unicode_script.cc
@@ -14,7 +14,8 @@
 
 #include <unordered_map>
 
-#include "third_party/absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_map.h"
+
 #include "unicode_script.h"
 #include "unicode_script_map.h"
 #include "util.h"
diff --git a/src/unicode_script_map.h b/src/unicode_script_map.h
index f2e67e9..6edfaa8 100644
--- a/src/unicode_script_map.h
+++ b/src/unicode_script_map.h
@@ -14,7 +14,9 @@
 
 #ifndef UNICODE_SCRIPT_DATA_H_
 #define UNICODE_SCRIPT_DATA_H_
-#include "third_party/absl/container/flat_hash_map.h"
+
+#include "absl/container/flat_hash_map.h"
+
 namespace sentencepiece {
 namespace unicode_script {
 namespace {
diff --git a/src/unicode_script_test.cc b/src/unicode_script_test.cc
index ab33565..76f2aa0 100644
--- a/src/unicode_script_test.cc
+++ b/src/unicode_script_test.cc
@@ -12,9 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.!
 
+#include "absl/strings/string_view.h"
+
 #include "common.h"
 #include "testharness.h"
-#include "third_party/absl/strings/string_view.h"
 #include "unicode_script.h"
 #include "util.h"
 
diff --git a/src/unigram_model.cc b/src/unigram_model.cc
index d9f1ce9..ee085ec 100644
--- a/src/unigram_model.cc
+++ b/src/unigram_model.cc
@@ -24,10 +24,11 @@
 #include <utility>
 #include <vector>
 
-#include "third_party/absl/container/flat_hash_map.h"
-#include "third_party/absl/memory/memory.h"
-#include "third_party/absl/strings/str_split.h"
-#include "third_party/absl/strings/string_view.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/memory/memory.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/unigram_model_test.cc b/src/unigram_model_test.cc
index bf22da3..567166a 100644
--- a/src/unigram_model_test.cc
+++ b/src/unigram_model_test.cc
@@ -19,11 +19,12 @@
 #include <string>
 #include <vector>
 
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_processor.h"
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_join.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/unigram_model_trainer.cc b/src/unigram_model_trainer.cc
index d58c408..baed626 100644
--- a/src/unigram_model_trainer.cc
+++ b/src/unigram_model_trainer.cc
@@ -24,13 +24,14 @@
 #include <utility>
 #include <vector>
 
+#include "absl/container/flat_hash_map.h"
+#include "absl/memory/memory.h"
+#include "absl/strings/str_replace.h"
+#include "absl/strings/str_split.h"
+
 #include "normalizer.h"
 #include "pretokenizer_for_training.h"
 #include "sentencepiece_trainer.h"
-#include "third_party/absl/container/flat_hash_map.h"
-#include "third_party/absl/memory/memory.h"
-#include "third_party/absl/strings/str_replace.h"
-#include "third_party/absl/strings/str_split.h"
 #include "third_party/esaxx/esa.hxx"  // Suffix array library.
 #include "unicode_script.h"
 #include "util.h"
diff --git a/src/unigram_model_trainer.h b/src/unigram_model_trainer.h
index c6562e6..0a3f640 100644
--- a/src/unigram_model_trainer.h
+++ b/src/unigram_model_trainer.h
@@ -20,8 +20,9 @@
 #include <utility>
 #include <vector>
 
+#include "absl/strings/string_view.h"
+
 #include "sentencepiece_model.pb.h"
-#include "third_party/absl/strings/string_view.h"
 #include "trainer_interface.h"
 #include "unigram_model.h"
 #include "util.h"
diff --git a/src/unigram_model_trainer_test.cc b/src/unigram_model_trainer_test.cc
index 9d2c526..ab887b6 100644
--- a/src/unigram_model_trainer_test.cc
+++ b/src/unigram_model_trainer_test.cc
@@ -17,13 +17,14 @@
 #include <string>
 #include <vector>
 
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+
 #include "filesystem.h"
 #include "sentencepiece_model.pb.h"
 #include "sentencepiece_processor.h"
 #include "sentencepiece_trainer.h"
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_join.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/util.h b/src/util.h
index 5110291..ece08e9 100644
--- a/src/util.h
+++ b/src/util.h
@@ -28,9 +28,10 @@
 #include <utility>
 #include <vector>
 
+#include "absl/strings/string_view.h"
+
 #include "common.h"
 #include "sentencepiece_processor.h"
-#include "third_party/absl/strings/string_view.h"
 
 #ifdef SPM_NO_THREADLOCAL
 #include <pthread.h>
diff --git a/src/util_test.cc b/src/util_test.cc
index 71d006f..e264081 100644
--- a/src/util_test.cc
+++ b/src/util_test.cc
@@ -14,9 +14,10 @@
 
 #include <map>
 
+#include "absl/strings/str_cat.h"
+
 #include "filesystem.h"
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
 #include "util.h"
 
 namespace sentencepiece {
diff --git a/src/word_model_trainer.cc b/src/word_model_trainer.cc
index 0b8b062..3f702fd 100644
--- a/src/word_model_trainer.cc
+++ b/src/word_model_trainer.cc
@@ -15,8 +15,9 @@
 #include <cmath>
 #include <string>
 
-#include "third_party/absl/container/flat_hash_map.h"
-#include "third_party/absl/strings/string_view.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/string_view.h"
+
 #include "util.h"
 #include "word_model.h"
 #include "word_model_trainer.h"
diff --git a/src/word_model_trainer_test.cc b/src/word_model_trainer_test.cc
index c4a8bc6..8288027 100644
--- a/src/word_model_trainer_test.cc
+++ b/src/word_model_trainer_test.cc
@@ -15,11 +15,12 @@
 #include <string>
 #include <vector>
 
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+
 #include "filesystem.h"
 #include "sentencepiece_processor.h"
 #include "testharness.h"
-#include "third_party/absl/strings/str_cat.h"
-#include "third_party/absl/strings/str_join.h"
 #include "util.h"
 #include "word_model_trainer.h"
 
